### ISSUE TITLE
"command >& filename"-redirect not recognized as error for POSIX shells

### ISSUE THREAD

**Issue by johnLate** [koalaman/shellcheck#736]
```
$ cat bad.sh 
#!/bin/sh
echo foo >& bar

$ shellcheck bad.sh; echo $?
0

$ dash bad.sh 
bad.sh: 3: bad.sh: Syntax error: Bad fd number
```

If I understand [POSIX](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_07_06) correctly, this syntax is only for duplicating output file descriptors — unlike bash, which uses `>&` and `&>` for redirection of stdout and stderr at the same time.

**Comment by koalaman** [comment]
Indeed. A warning has been added in 6a4b86c.

**Comment by hwdsl2** [comment]
Hello! The fix for this issue has unintended side effects. For example:

```
exiterr()  { echo "Error: ${1}" >&2; exit 1; }
                                ^-- SC2039: In POSIX sh, >& is undefined.
```

I think the above is valid POSIX for redirection to stderr. The warning is incorrect.

**Comment by koalaman** [comment]
Which version of shellcheck are you using? I'm not getting this error in the latest version (as given by shellcheck.net)

**Comment by hwdsl2** [comment]
@koalaman Thanks for the quick reply. The error occurs when the shebang is "/bin/sh":

```
#!/bin/sh

exiterr()  { echo "Error: ${1}" >&2; exit 1; }
```

**Comment by koalaman** [comment]
Good catch. This was due to a parsing error that considered `>&2` the same as `>& 2`. Fixed in 619b6c4
